daemon/import.go

```diff
@@ -60,15 +62,24 @@ func (daemon *Daemon) ImportImage(src string, repository, tag string, msg string
 		rc = inConfig
 	} else {
 		inConfig.Close()
+		if len(strings.Split(src, "://")) == 1 {
+			src = "http://" + src
+		}
 		u, err := url.Parse(src)
 		if err != nil {
 			return err
 		}
-		if u.Scheme == "" {
-			u.Scheme = "http"
-			u.Host = src
-			u.Path = ""
+
+		// Checking the file referenced by the URL exists to avoid sending
+		// status code 200 to the client although the download process will fail
+		resp, err = http.Head(u.String())
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode >= http.StatusBadRequest {
+			return fmt.Errorf("Unable to read file in %s: %s", u, resp.Status)
 		}
+
 		outStream.Write(sf.FormatStatus("", "Downloading from %s", u))
```

Comment:
It's not really downloading anything until the response is consumed though, right?